synapse/http/endpoint.py

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from twisted.internet.endpoints import SSL4ClientEndpoint, TCP4ClientEndpoint
+from twisted.internet.endpoints import HostnameEndpoint, wrapClientTLS
```

Comment:
It should probably be removed from there; `TCP4ClientEndpoint` is only interesting as a low-level implementation detail.